src/allmydata/storage_client.py

```diff
@@ -559,6 +559,7 @@ def from_announcement(cls, server_id, furl, ann, storage_server):
 
         *nickname* is optional.
         """
+        furl = furl.encode("utf-8")
```

Comment:
Maybe adding type docs to the docstring here would be good

Reply:
Ugggh I did it in the follow-up branch by mistake. So I'll just merge this and it'll be in the next PR, since it's not a significant change.